### ISSUE TITLE
Demanding RPC request fails

### ISSUE THREAD

**Issue by andrewheadricke** [ethereum/go-ethereum#17016]
Geth version: 1.8.11-stable
OS & Version: Linux

#### Expected behaviour
Return a list of events

#### Actual behaviour
Fails with error: 
```
ERROR[06-18|21:06:33] write tcp 127.0.0.1:8545->127.0.0.1:46102: i/o timeout
```

I have a js/web3 script which does an `event.get()` on a token address which returns potentially hundreds of thousands of events..  It used to work fine in both Geth and Parity.. however I just tried with geth 1.8.11 and geth is giving an error
```
ERROR[06-18|21:06:33] write tcp 127.0.0.1:8545->127.0.0.1:46102: i/o timeout
```
and causing the script to fail due to returning zero events.

The script still works fine using Parity.

Has some sort of timeout on long running queries been implemented?

**Comment by Beasta** [comment]
I am having the same problem when running a script. I get the i/o timeout in the geth logs and the script returns this:
```
Error: Invalid JSON RPC response: "" 
```

**Comment by ryanschneider** [comment]
Possibly related: https://github.com/ethereum/go-ethereum/pull/16880

What sort of timeouts are you seeing?  After 5~10 seconds, or after 2 minutes?

Can you include the output of `time curl ...` of the same query on a working pre-.11 node?

**Comment by andrewheadricke** [comment]
My query takes about 35-40 seconds, Geth is timing out at 20 seconds.

**Comment by Beasta** [comment]
The error for mine is very erratic and occurs sometimes quickly ( a few seconds), sometimes after a while, and sometimes not at all (takes ~30 seconds to complete).

I am running a large machine with >100 peers and have LightServ setting set to 90. Perhaps the inconsistency is due to varying amounts of LightServ activity?

**Comment by mikeshultz** [comment]
Sep 12 21:34:54 ip-w-x-y-z geth[16288]: ERROR[09-12|21:34:54.329] write tcp w.x.y.z:8080->a.b.c.d:52628: i/o timeout


Seen in 1.8.15 with `eth_getLogs` calls.  Size of expected response varies.  Really small block ranges will return results quickly but the range ends up being so small as to make this method almost unusable. 

If I hammer it repeatedly, it does actually seem to return data eventually, however. It's hard to pin down.  The same request sent repeatedly in succession will sometimes return and sometimes not without any noticeable pattern.  

As requested from @ryanschneider, here's some timed calls.  The response time doesn't always go down, but varies.  Generally they timeouts are between 40-50s.

    [mike@megan ~]$ time curl -w "%{http_code}" --data-binary '{"jsonrpc":"2.0","id":"666","method":"eth_getLogs","params": [{"address":"0x3818dC501b641905C83ebfF2ACd9f78fA6BD1341","fromBlock":"0x53ec60","toBlock":"latest"}]}' -H 'content-type:application/json' "https://mynode.net/";
    <html>
    <head><title>502 Bad Gateway</title></head>
    <body bgcolor="white">
    <center><h1>502 Bad Gateway</h1></center>
    </body>
    </html>
    502
    real    0m53.244s
    user    0m0.015s
    sys 0m0.012s
    [mike@megan ~]$ time curl -w "%{http_code}" --data-binary '{"jsonrpc":"2.0","id":"666","method":"eth_getLogs","params": [{"address":"0x3818dC501b641905C83ebfF2ACd9f78fA6BD1341","fromBlock":"0x53ec60","toBlock":"latest"}]}' -H 'content-type:application/json' "https://mynode.net/";
    <html>
    <head><title>502 Bad Gateway</title></head>
    <body bgcolor="white">
    <center><h1>502 Bad Gateway</h1></center>
    </body>
    </html>
    502
    real    0m46.457s
    user    0m0.022s
    sys 0m0.004s
    [mike@megan ~]$ time curl -s -w "%{http_code}" --data-binary '{"jsonrpc":"2.0","id":"666","method":"eth_getLogs","params": [{"address":"0x3818dC501b641905C83ebfF2ACd9f78fA6BD1341","fromBlock":"0x53ec60","toBlock":"latest"}]}' -H 'content-type:application/json' "https://mynode.net/";
    <html>
    <head><title>502 Bad Gateway</title></head>
    <body bgcolor="white">
    <center><h1>502 Bad Gateway</h1></center>
    </body>
    </html>
    502
    real    0m38.434s
    user    0m0.020s
    sys 0m0.007s

**Comment by mikeshultz** [comment]
Saw Peter pointed out a potential solution on Reddit today that you guys might be interested in.  I can't say why this started in a specific version(perhaps a performance issue?) but adjusting the write timeout allowed these requests to go through for me with 1.8.17(PR was merged for 1.8.13+).  

Even with the adjusted timeouts, the requests generally respond before the default timeout.  Though maybe that's just subjective observation with caching.

More details in this PR: https://github.com/ethereum/go-ethereum/pull/17240

**Comment by fjl** [comment]
BTW, you can use duration units in the config file :).
